spec/nylas/api_spec.rb

```diff
@@ -56,6 +56,59 @@
     end
   end
 
+  describe "#free_busy" do
+    it "returns `Nylas::FreeBusyCollection` for free busy details" do
+      emails = ["test@example.com", "anothertest@example.com"]
+      start_time = 1_609_439_400
+      end_time = 1_640_975_400
+      client = Nylas::HttpClient.new(
+        app_id: "not-real",
+        app_secret: "also-not-real",
+        access_token: "seriously-unreal"
+      )
+      api = described_class.new(client: client)
+      free_busy_response = [
+        {
+          object: "free_busy",
+          email: "test@example.com",
+          time_slots: [
+            {
+              object: "time_slot",
+              status: "busy",
+              start_time: 1_609_439_400,
+              end_time: 1_640_975_400
+            }
+          ]
+        }
+      ]
+      allow(client).to receive(:execute).with(
+        method: :post,
+        path: "/calendars/free-busy",
+        payload: {
+          emails: emails,
+          start_time: start_time,
+          end_time: end_time
+        }.to_json
+      ).and_return(free_busy_response)
+
+      result = api.free_busy(
+        emails: emails,
+        start_time: Time.at(start_time),
+        end_time: Time.at(end_time)
+      )
+
+      expect(result).to be_a(Nylas::FreeBusyCollection)
+      free_busy = result.last
```

Comment:
Is there a reason not to iterate over all of the objects in this array and test them? I'm assuming performance?

Reply:
First and last are different as one have time_slots and other do not have.